### DISCUSSION TITLE
macOS: Cannot connect to the Docker daemon

### DISCUSSION THREAD

**Discussion by jsjoeio** [coder/coder#4487] [General] [answered]
## Problem

While running `coder templates create` on macOS, I got this error:

```go
❯ coder templates create            
> Create and upload "~/dev/testing/coder-repro-2815"? (yes/no) 
✔ Queued [440ms]
✔ Setting up [64ms]
✔ Adding README.md... [62ms]
✔ Parsing template parameters [127ms]
⧗  Detecting persistent resources 
  Terraform 1.1.9
Error: Error pinging Docker server: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. 
Is the docker daemon running?
  
✔ Detecting persistent resources [1980ms]
✘ Cleaning Up [112ms]
template import provision for start: recv import provision: plan terraform: terraform plan: exit status 1
Run 'coder templates create --help' for usage. 
```
I have Docker Desktop running and it shows 
<img width="250" alt="image" src="https://user-images.githubusercontent.com/3806031/195151420-ac5af728-3bcf-4356-bd34-d45b89714fc3.png">

**Comment by jsjoeio** [comment]
I've tried the following solutions with no luck:
- https://stackoverflow.com/a/49719638
- https://stackoverflow.com/a/70294241

**Comment by bpmct** [comment]
What is the contents of `docker ps` on your terminal?

**Comment by bpmct** [comment, marked as answer]
@jsjoeio and I just chatted about this. The coder server (Linux VM) he was connected to via `coder login` didn't have a Docker socket running even though his client (macOS laptop) has Docker running. Workspaces are provisioned via the server but the raw Terraform output doesn't make that clear.

**Reply by jsjoeio** [reply]
To add to that, I restarted `coder server` to run it locally on my macOS and ti worked!
